pkg/foundation/metrics/metrics.go

```diff
@@ -405,14 +405,18 @@ func (mt *labeledHistogram) WithValues(vs ...string) Histogram {
 // RecordBytesHistogram wraps a histrogram metric and allows to observe record
 // sizes in bytes.
 type RecordBytesHistogram struct {
-	h Histogram
+	H Histogram
 }
 
 func NewRecordBytesHistogram(h Histogram) RecordBytesHistogram {
-	return RecordBytesHistogram{h}
+	return RecordBytesHistogram{H: h}
 }
 
 func (m RecordBytesHistogram) Observe(r opencdc.Record) {
+	m.H.Observe(m.SizeOf(r))
+}
+
+func (m RecordBytesHistogram) SizeOf(r opencdc.Record) float64 {
```

Comment:
This is exposed now so that sizes can be precalculated externally. Relevant for batch processing.